sync2/types/types_test.go

```diff
@@ -0,0 +1,51 @@
+package types
```

Comment:
As noted above, this will probably go into `rangesync` package.
Using pure black-box testing for packages requires exposing some internal implementation details for the sole purpose of verifying them in the tests. Not having tests for these parts just b/c they're internal (relying only on higher-level tests instead) is not really a good option as it may allow for some unpleasant bugs to go unnoticed, and will make debugging harder.
This can be fixed by using `internal` subpackages but that might mean more splitting things down to small parts than really necessary for the proper structure of the program, for the sole purpose of tests' convenience.
Will try it anyway, let's see

Reply:
You can use the `export_test.go` [hack](https://go.dev/src/os/export_test.go) for exporting things for the sake of testing. It's the lesser of evil afaik.

Reply:
Sounds reasonable, will do so, thanks

Reply:
Updated `sync2/rangesync` to use black-box testing